Content.Server/GameObjects/Components/Explosion/ClusterFlashComponent.cs

```diff
@@ -0,0 +1,186 @@
+using Content.Shared.Physics;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.Log;
+
+namespace Content.Server.GameObjects.Components.Explosives
+{
+    [RegisterComponent]
+    public sealed class ClusterFlashComponent : Component, IInteractUsing, IUse
+    {
+        public override string Name => "ClusterFlash";
+
+        private Container _grenadesContainer = default!;
+
+        /// <summary>
+        ///     What we fill our prototype with if we want to pre-spawn with grenades.
+        /// </summary>
+        [ViewVariables]
+        private string? _fillPrototype;
+
+        private int _unspawnedCount;
+
+        [ViewVariables]
+        private int _maxGrenades;
+
+        /// <summary>
+        ///     How long until our grenades are shot out and armed.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _delay;
+
+        /// <summary>
+        ///     Max distance grenades can be thrown.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _throwDistance;
+
+        /// <summary>
+        ///     This is the end.
+        /// </summary>
+        private bool _countDown;
+
+        // I'm suss on this bit
+        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs args)
+        {
+            if (_grenadesContainer.ContainedEntities.Count >= _maxGrenades || !args.Using.HasComponent<FlashExplosiveComponent>())
+                return false;
+
+            _grenadesContainer.Insert(args.Using);
+            UpdateAppearance();
+            return true;
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _fillPrototype, "fillPrototype", null);
+            serializer.DataField(ref _maxGrenades, "maxGrenadesCount", 3);
+            serializer.DataField(ref _delay, "delay", 1.0f);
+            serializer.DataField(ref _throwDistance, "distance", 3.0f);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _grenadesContainer = ContainerManagerComponent.Ensure<Container>("cluster-flash", Owner);
+
+        }
+
+        protected override void Startup()
+        {
+            base.Startup();
+
+            if (_fillPrototype != null)
+            {
+                _unspawnedCount = Math.Max(0, _maxGrenades - _grenadesContainer.ContainedEntities.Count);
+                UpdateAppearance();
+            }
+        }
+
+        bool IUse.UseEntity(UseEntityEventArgs eventArgs)
+        {
+            if (_countDown || ((_grenadesContainer.ContainedEntities.Count) <= 0 && (_fillPrototype == null)))
```

Comment:
Shouldn't this check for unspawned count instead?

Reply:
No, but we can change it to
```suggestion
            if (_countDown || (_grenadesContainer.ContainedEntities.Count + _unspawnedCount) <= 0))
```